cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,195 @@
+package methods
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/creachadair/jrpc2/handler"
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/go/toid"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/ledgerbucketwindow"
+)
+
+// TransactionsPaginationOptions defines the available options for paginating through transactions.
+type TransactionsPaginationOptions struct {
+	Cursor *toid.ID `json:"cursor,omitempty"`
+	Limit  uint     `json:"limit,omitempty"`
+}
+
+// GetTransactionsRequest represents the request parameters for fetching transactions within a range of ledgers.
+type GetTransactionsRequest struct {
+	StartLedger uint32                         `json:"startLedger"`
+	Pagination  *TransactionsPaginationOptions `json:"pagination,omitempty"`
+}
+
+// isValid checks the validity of the request parameters.
+func (req GetTransactionsRequest) isValid(maxLimit uint, ledgerRange ledgerbucketwindow.LedgerRange) error {
+	if req.Pagination != nil && req.Pagination.Cursor != nil {
+		if req.StartLedger != 0 {
+			return errors.New("startLedger and cursor cannot both be set")
+		}
+	} else if req.StartLedger < ledgerRange.FirstLedger.Sequence || req.StartLedger > ledgerRange.LastLedger.Sequence {
+		return errors.Errorf("start ledger must be between the oldest ledger: %d and the latest ledger: %d for this rpc instance.", ledgerRange.FirstLedger.Sequence, ledgerRange.LastLedger.Sequence)
+	}
+
+	if req.Pagination != nil && req.Pagination.Limit > maxLimit {
+		return fmt.Errorf("limit must not exceed %d", maxLimit)
+	}
+
+	return nil
+}
+
+// GetTransactionsResponse encapsulates the response structure for getTransactions queries.
+type GetTransactionsResponse struct {
+	Transactions          []db.Transaction `json:"transactions"`
```

Comment:
I don't think `db.Transaction` is the right representation to include in the response. The `db.Transaction` struct doesn't define any json tags on any of its fields

Reply:
@tamirms I was trying to avoid defining a duplicate struct and just reuse this. Since db.Transaction has exactly all the fields we want to return in the response.

Reply:
I understand but does the get transactions json response actually match the specification in the issue?

Reply:
I agree. It is better to add a separate struct